File: AcademicMarketplace/Modules/Services/workgroupService.ts
module AcademicMarketplace.Services {
    export class WorkgroupService {
        static $inject = ["$http"];

        private httpService: any;

        constructor(
            $http: any
        ) {
            this.httpService = $http;
        }

        getAll(): any {
            return this.httpService({
                method: "GET",
                url: "../Workgroup/GetAll/"
            });
        }

        addWorkgroup(model: Models.WorkgroupModel.IWorkgroupModel): any {
            return this.httpService({
                method: "POST",
                url: "../Workgroup/AddWorkgroup/",
                data: JSON.stringify(model),
                contentType: "application/json"
            });
        }

        editWorkgroup(model: Models.WorkgroupModel.IWorkgroupModel): any {
            return this.httpService({
                method: "PUT",
                url: "../Workgroup/EditWorkgroup/",
                data: JSON.stringify(model),
                contentType: "application/json"
            });
        }

        deleteWorkgroup(model: Models.WorkgroupModel.IWorkgroupModel): any {
            return this.httpService({
                method: "DELETE",
                url: "../Workgroup/DeleteWorkgroup/" + model.code
                //data: JSON.stringify(model),
                //contentType: "application/json"
            });
        }
    }
}

File: AcademicMarketplace/Business/WorkgroupService.cs
using System;
using System.Collections.Generic;
using AcademicMarketplace.Controllers.Models;
using AcademicMarketplace.Data;
using Microsoft.AspNet.Identity;


namespace AcademicMarketplace.Business
{

    public interface IWorkgroupService
    {
        List<WorkgroupModel> GetAll();
        List<WorkgroupModel> GetUserWorkgroups(UserModel user);
        WorkgroupModel AddWorkgroup(WorkgroupModel model);
        WorkgroupModel EditWorkgroup(WorkgroupModel model);
        string DeleteWorkgroup(string code);
    }

    public class WorkgroupService : IWorkgroupService
    {
        private IDataAccessService _data;

        public WorkgroupService()
        {
            _data = new DataAccessService();
        }

        public List<WorkgroupModel> GetAll()
        {
            return _data.GetAllWorkgroups();
        }

        public List<WorkgroupModel> GetUserWorkgroups(UserModel user)
        {
            var userObj = _data.GetUser(user.Id);
            return userObj.Workgroups;
        }

        public WorkgroupModel AddWorkgroup(WorkgroupModel model)
        {
            return _data.AddWorkgroup(model);
        }

        public WorkgroupModel EditWorkgroup(WorkgroupModel model)
        {
            return _data.EditWorkgroup(model);
        }

        public string DeleteWorkgroup(string code)
        {
            try
            {
                var attempt = _data.DeleteWorkgroup(code);
                return attempt;
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
            }
            return "fail";
        }
    }
}